src/main/groovy/org/transmartproject/batch/highdim/beans/AbstractStandardHighDimJobConfiguration.groovy

```diff
@@ -3,334 +3,25 @@ package org.transmartproject.batch.highdim.beans
-import org.transmartproject.batch.batchartifacts.FoundExitStatusChangeListener
-import org.transmartproject.batch.batchartifacts.LogCountsStepListener
-import org.transmartproject.batch.batchartifacts.PutInBeanWriter
-import org.transmartproject.batch.beans.AbstractJobConfiguration
 import org.transmartproject.batch.beans.JobScopeInterfaced
-import org.transmartproject.batch.clinical.db.objects.Tables
-import org.transmartproject.batch.concept.ConceptPath
-import org.transmartproject.batch.concept.DeleteConceptCountsTasklet
-import org.transmartproject.batch.concept.InsertConceptCountsTasklet
-import org.transmartproject.batch.concept.oracle.OracleInsertConceptCountsTasklet
-import org.transmartproject.batch.concept.postgresql.PostgresInsertConceptCountsTasklet
-import org.transmartproject.batch.db.DatabaseImplementationClassPicker
-import org.transmartproject.batch.db.DeleteByColumnValueWriter
-import org.transmartproject.batch.facts.ClinicalFactsRowSet
-import org.transmartproject.batch.facts.DeleteObservationFactTasklet
-import org.transmartproject.batch.facts.ObservationFactTableWriter
-import org.transmartproject.batch.highdim.assays.*
 import org.transmartproject.batch.highdim.datastd.*
-import org.transmartproject.batch.highdim.jobparams.StandardAssayParametersModule
 import org.transmartproject.batch.highdim.jobparams.StandardHighDimDataParametersModule
-import org.transmartproject.batch.highdim.platform.PlatformCheckTasklet
-import org.transmartproject.batch.highdim.platform.PlatformJobContextKeys
-import org.transmartproject.batch.highdim.platform.annotationsload.AnnotationEntity
-import org.transmartproject.batch.highdim.platform.annotationsload.AnnotationEntityMap
-import org.transmartproject.batch.patient.GatherCurrentPatientsReader
-import org.transmartproject.batch.patient.PatientSet
 import org.transmartproject.batch.support.JobParameterFileResource
 
 /**
- * A high dimensional job with standard assay mapping and data files.
- *
- * May need to be further broken down in the future.
+ * Standard HD configuration implementation that use {@link TripleStandardDataValue} as item
  */
-@ComponentScan(['org.transmartproject.batch.highdim.datastd',
-                'org.transmartproject.batch.highdim.compute',
-                'org.transmartproject.batch.highdim.assays',
-                'org.transmartproject.batch.highdim.i2b2',
-                'org.transmartproject.batch.concept',
-                'org.transmartproject.batch.patient', ])
-@SuppressWarnings('MethodCount')
-abstract class AbstractStandardHighDimJobConfiguration extends AbstractJobConfiguration {
-
-    public static final int LOAD_ANNOTATION_CHUNK_SIZE = 5000
-    public static final int LOAD_PATIENTS_CHUNK_SIZE = 512
-    public static final int DELETE_DATA_CHUNK_SIZE = 50
-    public static final int WRITE_ASSAY_CHUNK_SIZE = 50
+abstract class AbstractStandardHighDimJobConfiguration extends AbstractHighDimJobConfiguration {
```

Comment:
This is going too far. You now have to go through an hierarchy of 4 classes to understand the most basic thing about a job -- the steps it's composed of. The previous `AbstractStandardHighDimJobConfiguration` already left me a bit uncomfortable\*, but the amount delegated to the concrete was still small enough that we could mostly follow the job steps by looking just at `AbstractStandardHighDimJobConfiguration`.

Please explore alternatives not involving inheritance, if necessary violating DRY principles.

\* Not just because of the complexity and jumping around required, but because abstract classes constrain the parameters of the subclasses and from there limit the beans that can be injected into the method).